### DISCUSSION TITLE
Beta campaign (Beta)

### DISCUSSION THREAD

**Discussion by theloveroflife** [endless-sky/endless-sky#10829] [Ideas]
I need help finishing up my plugin for a beta campaign about halfway through your first year in a save, after remembrance day here is the code:
[characters.txt](https://github.com/user-attachments/files/17966603/characters.txt)
[factions.txt](https://github.com/user-attachments/files/17966604/factions.txt)
[missions.txt](https://github.com/user-attachments/files/17966605/missions.txt)
[outfits.txt](https://github.com/user-attachments/files/17966606/outfits.txt)
[sales.txt](https://github.com/user-attachments/files/17966607/sales.txt)
[ships.txt](https://github.com/user-attachments/files/17966608/ships.txt)
[systems_planets.txt](https://github.com/user-attachments/files/17966609/systems_planets.txt)
please note any problems, inconsistencies, or stupid add Ons, tell me and I will fix it. Once it has been finalized, I will share it on the plugin list for downloading, hopefully.

**Comment by TheGiraffe3** [comment]
It looks like you're mostly writing that in Naev's style, though good job avoiding making the missions file entirely Lua.
Try reading the wiki and the game files for more concrete examples of how ES' data format works.
At the moment, your campaign is unplayable because of `on_complete` instead of `"on complete"` and things like that.

**Comment by TheGiraffe3** [comment]
I would recommend that you create a GitHub repository and put those files in it so that it is easier for people to see the files (and then they can directly contribute).
